### PR TITLE
Sw 348 copy sw52 components

### PR DESCRIPTION
**Pull request by zaifastafa**

### CODE REVIEW

File: FinSearchUnified/Tests/Bundle/SearchBundle/CriteriaRequestHandler/FindologicFacetCriteriaRequestHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+        $condition = $criteria->getConditions();
+        if ($hasCondition === false) {
+            $this->assertEmpty($condition);
+        }
+        if ($hasCondition === true) {
+            $this->assertNotEmpty($condition);
+            $this->assertTrue($criteria->hasCondition('product_attribute_' . $field));
+            $condition = $criteria->getCondition('product_attribute_' . $field);
+            $conditionField = $condition->getField();
```
Comment (dblaichinger) [inline FinSearchUnified/Tests/Bundle/SearchBundle/CriteriaRequestHandler/FindologicFacetCriteriaRequestHandlerTest.php]: As stated in the story: "Passed criteria object has a condition of type ProductAttributeCondition"
Please assert that `FinSearchUnified/Bundle/SearchBundle/Condition/ProductAttributeCondition` is used, so we are sure not the default Shopware ProductAttributeCondition is used.

File: FinSearchUnified/Bundle/SearchBundle/CriteriaRequestHandler/FindologicFacetCriteriaRequestHandler.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+use FinSearchUnified\Bundle\SearchBundle\Condition\Operator;
+use FinSearchUnified\Bundle\SearchBundle\Condition\ProductAttributeCondition;
+use FinSearchUnified\Bundle\StoreFrontBundle\Service\CustomFacetServiceInterface;
+use FinSearchUnified\Helper\StaticHelper;
+use Shopware\Bundle\SearchBundle\Condition\CombinedCondition;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\CriteriaRequestHandlerInterface;
+use Shopware\Bundle\SearchBundle\Facet\CombinedConditionFacet;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\StoreFrontBundle\Struct\Search\CustomFacet;
```
[the remark this answers is not in the document]
Comment (zaifastafa) [reply FinSearchUnified/Bundle/SearchBundle/CriteriaRequestHandler/FindologicFacetCriteriaRequestHandler.php]: yes this is part of the next story. like I have mentioned in the youtrack comment, I had fixed these issues while doing code review, but then I saw that the SW-349 story was about implementing these changes only, so I did not push those changes and will let Rabab implement them.

File: FinSearchUnified/Tests/Bundle/SearchBundle/Condition/ProductAttributeConditionTest.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     * @param string $field
+     * @param string $exception
+     */
+    public function testConditionException($field, $exception)
+    {
+        if ($exception !== null) {
+            $this->expectException($exception);
+        }
+
+        $condition = new ProductAttributeCondition($field, Operator::EQ, 'Findologic');
```
[the remark this answers is not in the document]
Comment (zaifastafa) [reply FinSearchUnified/Tests/Bundle/SearchBundle/Condition/ProductAttributeConditionTest.php]: ahan, nice! :+1:

**Comment by zaifastafa** [comment]
> ## Review
> * Some small issues to fix, but nothing major
> * Build is broken, you mentioned it will be handled in SW-349? So there is no possibility to fix them within this pull request?

Yes it can be fixed, but there's a separate story for it. After that the build will pass.

**Pull request merged**